rholang/src/main/scala/coop/rchain/rholang/build/CompiledRholangSource.scala

```diff
@@ -34,8 +35,11 @@ object CompiledRholangSource {
   * @param env a sequence of pairs macro -> value
   * @return
   */
-abstract class CompiledRholangTemplate(classpath: String, env: (String, Any)*)
-    extends CompiledRholangSource {
+abstract class CompiledRholangTemplate(
+    classpath: String,
+    deployerPk: Option[PublicKey],
```

Comment:
Why is this an option? Can't you make it mandatory?

Reply:
Take a look at which `CompiledRholangSource`s actually take some public key. It is just one out of many.
And I think it is valuable to make it explicit which rholang needs the public key and which doesn't for at least debugging reasons.

Reply:
I'm thinking all the Rholang deployed should need a deployer identity but then we would have a few predefined ones like genesis identity and tester identity.

This would help us get rid of the Option above and will simplify the logic

Reply:
That's a good point! I'd leave this for a separate PR though. It seems that CompliedRholangSource has less and less sense on its own, and needs to pull in basicaly the whole DeployData? I'd like to see how this evolves.

Reply:
I still think having defaults could make it harder to debug rholang where deployer/deploy ids are involved. Happy to create a ticket though if I am overruled :) Should I?

Reply:
I'm ok with a separate PR for this